main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooser.java

```diff
@@ -0,0 +1,170 @@
+ * </ul>
+ * If the {@code #prepare(Path)}-Method of a MPC fails the entire Mountpoint-Choosing-Operation
+ * is aborted and the method should do all necessary cleanup before throwing the exception.
+ * If the preparation succeeds {@link #cleanup(Path)} can be used after unmount to do any
+ * remaining cleanup.
+ */
+public interface MountPointChooser {
+
+	/**
+	 * Called by the {@link MountPointChooserModule} to determine whether this MountPointChooser is
+	 * applicable for the given Systemconfiguration.
+	 *
+	 * <p>The result of this method defaults to true. Developers should override this method to
+	 * check for Systemconfigurations that are unsuitable for this MPC.
+	 *
+	 * @return a boolean flag; true if applicable, else false.
+	 * @see #chooseMountPoint()
+	 */
+	default boolean isApplicable() {
+		return true; //Usually most of the choosers should be applicable
+	}
+
+	/**
+	 * Called by a {@link Volume} to do choose a Mountpoint according to the
+	 * MountPointChoosers strategy.
+	 *
+	 * <p>This method is only called for MPCs that were deemed {@link #isApplicable() applicable}
+	 * by the {@link MountPointChooserModule MountPointChooserModule.}
+	 * Developers should override this method to find or extract a Mountpoint for
+	 * the Volume <b>without</b> preparing it. Preparation should be done by
+	 * {@link #prepare(Path)} instead.
+	 * Exceptions in this method should be handled gracefully and result in returning
+	 * {@link Optional#empty()} instead of throwing an exception.
+	 *
+	 * @return the chosen path or {@link Optional#empty()} if an exception occurred
+	 * or no Mountpoint could be found.
+	 * @see #isApplicable()
+	 * @see #prepare(Path)
+	 */
+	Optional<Path> chooseMountPoint();
+
+	/**
+	 * Called by a {@link Volume} to prepare and/or verify the chosen Mountpoint.<br>
+	 * This method is only called if the {@link #chooseMountPoint()}-Method of the same
+	 * MountPointChooser returned a path.
+	 *
+	 * <p>Developers should override this method to prepare the Mountpoint for
+	 * the Volume and check for any obstacles that could hinder the Mount-Operation.
+	 * The Mountpoint is deemed "prepared" if it can be used to mount a Volume
+	 * without any further Filesystemactions or Userinteraction. If this is not possible,
+	 * this method should fail. In other words: This method should not return without
+	 * either failing or finalizing the preparation of the Mountpoint.
+	 * Generally speaking exceptions should be wrapped as
+	 * {@link InvalidMountPointException} to allow efficient handling by the caller.
+	 *
+	 * <p>Often the preparation of a Mountpoint involves creating files or others
+	 * actions that require cleaning up after the Volume is unmounted.
+	 * In this case developers should override the {@link #cleanup(Path)}-Method
+	 * and return {@code true} to the Volume to indicate that the
+	 * {@code #cleanup}-Method of this MPC should be called after unmount.
```

Comment:
If it is already mentioned there, it can be deleted in the javadoc for the `prepare()` method. (; Otherwise it needs to maintained, if something changes in the interface.

Reply:
But as I said: The following paragraph depends on this information, also the wording differs between `#prepare()` and `#cleanup()` to allow stressing important parts. I see how this would increase the total maintanance needed, on the other hand, it's far easier to understand that way.

Reply:
Resolved in personal conversation.